webplugin/js/app/kommunicate.js

```diff
@@ -790,29 +790,16 @@ $applozic.extend(true, Kommunicate, {
         return true;
     },
     hideMessage: function (element) {
-        if (!element) {
-            return;
-        }
-        var parentEle = element.parentElement;
-        while (!parentEle.classList.contains('mck-msg-left')) {
-            parentEle = parentEle.parentElement;
-        }
-        parentEle.classList.add('n-vis');
-
-        var quickReplyCtaPrevSibling = parentEle.previousElementSibling;
+        //all cta expect link button will hide if hidePostCta is enable
+        if (!element && !element[0]) return;
+        var quickReplyCtaPrevSibling = element[0];
 
-        while (
-            quickReplyCtaPrevSibling &&
-            quickReplyCtaPrevSibling.classList.contains(
-                'contains-quick-replies-only'
-            )
-        ) {
-            quickReplyCtaPrevSibling =
-                quickReplyCtaPrevSibling.previousElementSibling;
-        }
-        if (quickReplyCtaPrevSibling) {
-            quickReplyCtaPrevSibling.classList.remove('km-clubbing-first');
+        while (quickReplyCtaPrevSibling.classList.contains('contains-quick-replies-only')) {
```

Comment:
Can we rely on this class?
Please check for all rich messages 
As far as I remember there is some difference in the logic for appending this class in different rich messages

Reply:
Reference:
![Screenshot 2023-05-15 at 11 43 56 AM](https://github.com/Kommunicate-io/Kommunicate-Web-SDK/assets/66241555/1a756f0c-3e30-4f67-905a-7ef2130b4189)